packages/ra-no-code/src/ResourceConfiguration/ResourceConfigurationProvider.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { ReactNode, useCallback, useMemo, useState } from 'react';
+import {
+    ResourceConfigurationMap,
+    ResourceConfiguration,
+    ResourceConfigurationContext,
+} from './ResourceConfigurationContext';
+
+export const ResourceConfigurationProvider = ({
+    children,
+}: {
+    children: ReactNode;
+}) => {
+    const [resources, setInternalResources] = useState<
+        ResourceConfigurationMap
+    >(() => loadConfigurationsFromLocalStorage());
+
+    const setResources = useCallback(
+        (
+            value: (
+                prevState: ResourceConfigurationMap
+            ) => ResourceConfigurationMap
+        ) => {
+            setInternalResources(prevState => {
+                const newState = value(prevState);
+
+                if (newState != undefined) { // eslint-disable-line
+                    window.localStorage.setItem(
+                        StorageKey,
```

Comment:
why the Capital S? It's a variable name, not a type or a component.

Reply:
It's a constant

Reply:
The make it ALL_CAPS

Reply:
Made it a prop with a default to the constant properly cased